### ISSUE TITLE
[WalletConnect] Upgrade to WalletConnect v2

### ISSUE THREAD

**Issue by dasanra** [safe-global/safe-react-apps#178]
## Overview
WalletConnect v2 was released in the past days. It may worth to start creating a branch using the latest version of the dapp connector, as they expect to deprecate v1.
Estimated time for deprecation will be 6 months after v2 use surpass v, so no specific date for shutdown

## Goals
We add compatibility with WalletConnect v2 so we don't get chased by the v1 shutdown

## Requirements
 - Upgrade to WalletConnect v2
 - We have to test with other apps, specially if it's backwards compatible with Wallet Connect v2. We don't know if connection will work as expected with apps using an older version as connector.

## Links
In their docs they say that Wallets can detect to route v1 or v2 so it makes sense that we push into upgrade this faster in the Safe app
https://docs.walletconnect.com/migration#migration-schedule

**Comment by mikheevm** [comment]
After working on the ticket for a while I suggest we postpone the upgrade because implementing it now involves a lot of friction:
1. The documentation is VERY immature, with no code examples whatsoever
2. WalletConnect team doesn't actively answer questions on discord
3. It seems that they haven't tested simultaneous integration of v1 and v2 very well, for example, I had to contribute to the types package because types v1 overwrote types for v2 when using them at the same time.
4. There is no time pressure to do it even in the next following months

cc @rmeissner @dasanra @sche

**Comment by dasanra** [comment]
@mikheevm thank you for your research. Will add a monthly alert for myself to check the state of WalletConnect v2 later so we don't find that suddenly the change is mandatory

**Comment by pedrouid** [comment]
Thanks @mikheevm for the feedback and your PR 🙌 

Our documentation has a long way to go and we should have pointed out to code examples that we have in our monorepo that would have made it clearer for developers to integrate WalletConnect v2.0

https://github.com/WalletConnect/walletconnect-monorepo/tree/v2.0/examples

In the link above you can see the directory where have updated examples for both a React-based wallet and dapp using WalletConnect v2.0 with the latest beta version

Our intention is that all wallets integrate WalletConnect v1.0 and v2.0 in parallel and we received multiple reports of the dependency conflicts observed by early integrators of these packages. So we now have a new NPM package that be integrated in parallel without using NPM aliases or requiring type declaration fixes.

```
npm install --save @walletconnect/client @walletconnect/legacy-client
```

While there isn't necessarily time pressure from the Dapp side to integrate v2.0 it's important that Wallets integrate it upfront to provide enough timeframe for Dapps to comfortably migrate since they don't get the option of supporting both versions in parallel. We have written about this in our migration schedule:

https://docs.walletconnect.com/2.0/migration

I'm grateful that Gnosis Safe takes the opportunity to be an early adopter of the WalletConnect v2.0 and we have created a group chat on Discord with your team members to make async communication easier

Please let me know of more feedback you might have for us so that we can WalletConnect v2.0 adoption easier for everyone including yourselves

**Comment by mikheevm** [comment]
> Our documentation has a long way to go and we should have pointed out to code examples that we have in our monorepo that would have made it clearer for developers to integrate WalletConnect v2.0

https://github.com/WalletConnect/walletconnect-monorepo/tree/v2.0/examples

Yep, I've seen these repos but what I meant is more a migration guide doc, I found it very helpful from my experience when there's a dedicated document telling you what you need to change in order to migrate to the newer version. With the current approach, we'd have to spend time reading wallet/dapp code to understand how we should do it.

cc'ing @dasanra and @sche for another re-evaluation

**Comment by dasanra** [comment]
@clemoon this has been waiting to development to go further as we didn't had capacity to keep on the beta stage

Now they are on release candidate it's much more likely that we should start checking it soon.
https://github.com/WalletConnect/walletconnect-monorepo/releases

**Comment by clemoon** [comment]
![CleanShot 2022-11-13 at 17 07 22@2x](https://user-images.githubusercontent.com/21241679/201611033-3b479d2a-ee43-467d-ad1a-de61862963d4.png)
We should also update the Safe Apps description of WalletConnect as pointed by Lukas. Apps which have updated directly to V2 offer "Copy to clipboard" by default

**Comment by topdev658** [comment]
Hi everyone!
I am trying walletconnect v2 now.  But it isn't working on my windows 10 64 bit.

I got errors when npm build after npm install.

How can I lunch v2 on my windows 10 64 bit.  Please help me.
